### PR TITLE
Block access to Metabase login uris

### PR DESCRIPTION
**Pull request by ogusak**
## Description

Restrict access to Metabase from public internet

## Type of change
- [ ] Bug fix
- [x] New feature
- [ ] Breaking change

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -646,3 +646,15 @@ variable "cloudwatch_agent_kong_config" {
   # set non-existent parameter name to avoid granting broad permissions
   default = "non-existent-parameter"
 }
+
+variable "external_lb_paths_deny_on_post" {
+  description = "List of path to deny access from public internet using POST method "
+  type        = list(string)
+
+  default = [
```
Comment (tovbinm) [inline variables.tf]: `default = []`

File: lb.tf [hunk 2]
```diff
@@ -84,11 +84,39 @@ resource "aws_lb_listener" "external-https" {
   }
 }
 
+resource "aws_lb_listener_rule" "deny_post_to_paths" {
+  count        = var.enable_external_lb ? 1 : 0
```
Comment (tovbinm) [inline lb.tf]: `count = var.enable_external_lb && length(var.external_lb_paths_deny_on_post) > 0 ? 1 : 0`

File: lb.tf [hunk 3]
```diff
@@ -84,11 +84,39 @@ resource "aws_lb_listener" "external-https" {
...
+
+  condition {
+    path_pattern {
+      values = var.external_lb_paths_deny_on_post
+    }
+  }
+
+  condition {
+    http_request_method {
+      values = ["POST"]
```
Comment (tovbinm) [inline lb.tf]: `values = var.external_lb_paths_deny_methods`

File: lb.tf [hunk 4]
```diff
@@ -84,11 +84,39 @@ resource "aws_lb_listener" "external-https" {
...
+    fixed_response {
+      content_type = "text/plain"
+      message_body = "Forbidden"
+      status_code  = "403"
+    }
+  }
+
+  condition {
+    path_pattern {
+      values = var.external_lb_paths_deny_on_post
```
Comment (tovbinm) [inline lb.tf]: `values = var.external_lb_paths_deny_paths`

File: lb.tf (hunk 3 above)
Comment (ogusak) [reply lb.tf]: i thought about that - it becomes a bit convoluted as all these conditions are `AND` - for instance - for metabase we do not want to block GET requests on the same uri

Comment (tovbinm) [reply lb.tf]: This is a generic module, so we should try to keep it this way. Any other specific settings should be passed as vars.

**Review by tovbinm — APPROVED** [review]

**Pull request merged**